src/components/utils/groupedRegions.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import { Region } from '../../xpanse-api/generated';
+
+export interface GroupedRegionItem {
+    /**
+     * The names of the region
+     */
+    name: string[];
+    /**
+     * The sites with the region belongs to, such as default, Chinese Mainland, International,
+     */
+    site: string[];
+    /**
+     * The area which the region belongs to, such as Asia, Europe, Africa
+     */
+    area: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+// const groupedRegions1 = (regions: Region[]): GroupedRegionItem[] => {
+//     const groupedData: Record<string, GroupedRegionItem> = {};
+//
+//     regions.forEach((item) => {
+//         // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+//         if (!groupedData[item.area]) {
+//             groupedData[item.area] = {
+//                 name: [item.name],
+//                 site: [item.site],
+//                 area: item.area,
+//             };
+//         } else {
+//             // if(!groupedData[item.area].name)
+//             groupedData[item.area].name.push(item.name);
+//             groupedData[item.area].site.push(item.site);
+//         }
+//     });
+//
+//     // Convert the grouped object back to an array
+//     return Object.values(groupedData);
+// };
+
+export const groupedRegions = (regions: Region[]): GroupedRegionItem[] => {
+    const groupedData: Record<string, GroupedRegionItem> = {};
+
+    regions.forEach((item) => {
+        // Create a composite key using both name and area
+        const groupKey = `${item.name}-${item.area}`;
+
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
try using any other condition to check if the key exists in the map. Maybe then we can avoid this comment.